pkg/console/install_panels.go

```diff
@@ -1285,6 +1289,61 @@ func addHostnamePanel(c *Console) error {
 	return nil
 }
 
+// ParseMask parses s as an IP mask, returning the result.
+// The string s must be in IPv4 dotted decimal form ("255.255.255.0").
+// Each octet must be a valid number between 0 and 255, and the bits must be
+// continuous (all 1s followed by all 0s when viewed as a 32-bit number).
+// If s is not a valid textual representation of an IP mask,
+// ParseMask returns an error describing why the parsing failed.
+func ParseMask(mask string) (IPMask, error) {
+    // Split the mask string by dots
+    parts := strings.Split(mask, ".")
+
+    // Validate number of parts
+    if len(parts) != 4 {
+           return nil, &ParseError{Type: "invalid mask format: must be x.x.x.x", Text: mask}
+    }
+
+    result := make(IPMask, 4)
+
+    // Parse and validate each octet
+    for i, part := range parts {
+        // Convert string to integer
+        num, err := strconv.Atoi(part)
+        if err != nil {
+               return nil, &ParseError{Type: fmt.Sprintf("invalid number in position %d: %s", i+1, part), Text: mask}
+        }
+        // Validate range (0-255)
+        if num < 0 || num > 255 {
+               return nil, &ParseError{Type: fmt.Sprintf("number out of range in position %d: %d", i+1, num), Text: mask}
+        }
+
+        result[i] = byte(num)
+    }
+
+    // Validate mask format (must be continuous 1s followed by continuous 0s)
+    if !isValidMaskFormat(result) {
+           return nil, &ParseError{Type: "invalid subnet mask: not continuous", Text: mask}
+    }
+
+	return result, nil
+}
+
+// Helper function to validate mask format
+func isValidMaskFormat(mask IPMask) bool {
+    var valid bool
+    outMask := net.IPv4Mask(mask[0], mask[1], mask[2], mask[3])
+    if outMask != nil {
+        ones, _ := outMask.Size()
+        cidrMask := net.CIDRMask(int(ones), 32)
+        valid = outMask.String() == cidrMask.String()
+    } else {
```

Comment:
If `valid := false` is set above, then the whole “else” branch is superfluous.
```suggestion
    }
```

Reply:
Thanks @votdev . I've fixed it based on you suggestion.